tests/lang/s01_basics/s08_user_defined_data_types/t02_plain_objects.nim

```diff
@@ -0,0 +1,146 @@
+discard """
+description: '''
+Covers basic features of objects - definition, construction, field access, field
+modification, copying.
+'''
+"""
+
+block simple_object:
+  ## Object can be defined in a `type` section, using `TypeName = object` 
+  ## and then listing all fields 
+  type
+    Simple = object
+      field1: int 
+      field2: int
+
+  ## Object can be constructed using `TypeName(field: value, ...)` syntax.
+  ## Fields that were not explicitly listed are set to their default values.
+  ## To assert that particular field is explicitly initialized in constructor
+  ## you can annotate it with `{.requiresinit.}` (for example of this see 
+  ## `t02_plain_object_requiresinit.nim`)
+  let it = Simple(field1: 12)
+
+  ## To get value of a particular field use `value.field` syntax
+  let field1Value = it.field1
+
+  doAssert field1Value == 12
+  doAssert it.field1 == 12
+
+  ## Fields that were not explicitly listed are set to the default values
+  doAssert it.field2 == 0
+
+  ## Order of field initialization is not important
+  let it1 = Simple(field1: 2, field2: 3)
+  let it2 = Simple(field2: 3, field1: 2)
+
+  doAssert it1.field1 == it2.field1
+  doAssert it2.field2 == it2.field2
+
+block copying_defaulting:
+  ## It is possible to create an object variable without explicitly initializing
+  ## it. In that case all fields would be initialized to default state (just like 
+  ## explicit initialization where omitted fields were defaulted)
+  type
+    Object = object
+      field1: int
+      field2: int
+
+  var defaulted: Object
+
+  doAssert defaulted.field1 == 0
+  doAssert defaulted.field2 == 0
+
+
+  ## Object can be copied into another variable
+  var copy = defaulted
+
+  ## Modifying copy does not affect original value
+  copy.field1 = 12
+
+  doAssert copy.field1 == 12
+  doAssert defaulted.field1 == 0
+
+
+
+block no_fields:
+  ## It is possible to define object with no fields. 
+  type
+    NoFields = object 
+
+  let it = NoFields()
+
+block using_complex_types_in_fields:
+  ## It is possible to define multiple objects in the same type section.
+  type
+    OtherPre = object
+      field1: int
+      field2: int
+ 
+    Object = object
+      ## It is possible to put any type as a field value *except* for the same
+      ## objet type *unless* it was defined as `ref object`. For example of this 
+      ## see `t02_plain_object_illegal_recursion.nim`
+      intField: int
+
+      anonTuple: (int, int)
+
+      objectPre: OtherPre
+      objectPost: OtherPost
+
+      araryField: array[5, int]
+
+    OtherPost = object
+      field1: int
+      field2: int
+
+block editing_object:
```

Comment:
Writing/mutating object instances

Sounds better, editing objects sounds like it might be structural